File: 02. Algorithms/01. Arrays/04. 75. Sort Colors OR Sort012/Solution.cpp
// https://leetcode.com/problems/sort-colors/
class Solution {
public:
    void sortColors(vector<int>& a) {
        int n = a.size();
        int low=0, mid=0, high=n-1;
        while(mid<=high) {
            if(a[mid]==0)
            {
                swap(a[mid++],a[low++]);
            }
            else if(a[mid]==1)
            {
                mid++;
            }
            else
            {
                swap(a[mid],a[high--]);
            }
        }
    }
};

File: 02. Algorithms/01. Arrays/04. 75. Sort Colors OR Sort012/solution.js
// problem link https://leetcode.com/problems/sort-colors
// submission link https://leetcode.com/problems/sort-colors/submissions/852815202/

function sortColors(nums) {
 
 let i = 0;
 let l = 0;
 let r = nums.length - 1;

 while(i <= r) {
     const num = nums[i];
     if(num ===  0) {
         swap(nums,i,l);
         i++;
         l++;
     } else if(num === 2) {
         swap(nums,i,r);
         r--;
     } else {
         i++;
     }
 } 

 return nums;
}

function swap(nums,i,j) {
    [nums[i], nums[j]] = [nums[j],nums[i]];
}
